omise/views/templates/hook/1.6/card_payment.tpl

```diff
@@ -77,63 +77,49 @@
   // the JS work correctly when the checkout is in one-page mode. It would appear that
   // dynamically created script blocks do not run in the global context
 
+  window.omiseCardFields = [
+    'name',
+    'number',
+    'expiration_month',
+    'expiration_year',
+    'security_code'
+  ];
+
   window.omiseCheckout = function omiseCheckout() {
     if (typeof Omise === 'undefined') {
       alert('{l s='Unable to process the payment, loading the external card processing library is failed. Please contact the merchant.' mod='omise'}');
       return false;
     }
 
-    omiseLockCheckoutForm(omiseCheckoutForm);
+    omiseSetCardFormLockedState(omiseCheckoutForm, true);
 
-    const card = {
-      name: omiseCheckoutForm.name.value,
-      number: omiseCheckoutForm.number.value,
-      expiration_month: omiseCheckoutForm.expiration_month.value,
-      expiration_year: omiseCheckoutForm.expiration_year.value,
-      security_code: omiseCheckoutForm.security_code.value,
-    };
+    const card = window.omiseCardFields.reduce(function(obj, field) {
+      obj[field] = omiseCheckoutForm[field].value;
+      return obj;
+    }, {});
 
     Omise.setPublicKey('{$omise_public_key}');
     Omise.createToken('card', card, omiseCreateTokenCallback);
-  }
+  };
 
   window.omiseCreateTokenCallback = function omiseCreateTokenCallback(statusCode, response) {
     if (statusCode === 200) {
       document.getElementById('omise_card_token').value = response.id;
       document.getElementById('omise_checkout_form').submit();
     } else {
       alert(response.message);
-      omiseUnlockCheckoutForm(omiseCheckoutForm);
+      omiseSetCardFormLockedState(omiseCheckoutForm, false);
     }
   };
 
-  window.omiseCheckoutForm = {
-    name: document.getElementById('omise_card_holder_name'),
-    number: document.getElementById('omise_card_number'),
-    expiration_month: document.getElementById('omise_card_expiration_month'),
-    expiration_year: document.getElementById('omise_card_expiration_year'),
-    security_code: document.getElementById('omise_card_security_code'),
-    checkout_button: document.getElementById('omise_checkout_button'),
-    checkout_text: document.getElementById('omise_checkout_text'),
-  };
+  window.omiseCheckoutForm = window.omiseCardFields.concat(['checkout_button', 'checkout_text']).reduce(function(obj, field) {
+    obj[field] = document.getElementById('omise_card_' + field);
+    return obj;
+  }, {});
 
-  window.omiseLockCheckoutForm = function omiseLockCheckoutForm(form) {
-    form.name.disabled = true;
-    form.number.disabled = true;
-    form.expiration_month.disabled = true;
-    form.expiration_year.disabled = true;
-    form.security_code.disabled = true;
-    form.checkout_button.disabled = true;
-    form.checkout_text.innerHTML = '{l s='Processing' mod='omise'}';
-  };
+  window.omiseSetCardFormLockedState = function omiseSetCardFormLockedState(form, state) {
```

Comment:
I had a very long long time relationship with JS and it's not quite up-to-date.
Wonder do we need to declare function name again? 
Can it be `window.omiseSetCardFormLockedState = function(form, state) { ... }`?

Reply:
It was like that already. I believe it assists in debug trace. If you don't give a name to the function, they just appear anonymously in the stack